pkg/planner/core/plan_clone_utils.go

```diff
@@ -140,6 +140,9 @@ func cloneConstantsForPlanCache(constants, cloned []*expression.Constant) []*exp
 	}
 	allSafe := true
 	for _, c := range constants {
+		if c == nil {
```

Comment:
why it's nil here

Reply:
In cases like "where pk in (?, ?, 1)", then this constant can be nil: https://github.com/pingcap/tidb/blob/master/pkg/planner/core/point_get_plan.go#L1160